shared/common-adapters/oriented-image.desktop.js

```diff
@@ -102,79 +164,100 @@ class OrientedImage extends React.Component<Props, State> {
 
   // Read the file contents directly into a buffer and pass it to EXIF which
   // can extract the EXIF data
-  _readExifLocal = src => {
+  _readExifLocal = () => {
+    const {src} = this.props
     return new Promise((resolve, reject) => {
-      try {
-        // data is a Node Buffer which is backed by a JavaScript ArrayBuffer.
-        // EXIF.readFromBinaryFile takes an ArrayBuffer
-        const data = fs.readFileSync(src)
-        const tags = EXIF.readFromBinaryFile(data.buffer)
-        tags ? resolve(tags['Orientation']) : reject(new Error('EXIF failed to read exif data'))
-      } catch (err) {
-        reject(err)
-      }
+      // data is a Node Buffer which is backed by a JavaScript ArrayBuffer.
+      // EXIF.readFromBinaryFile takes an ArrayBuffer
+      const data = fs.readFileSync(src)
+      const tags = EXIF.readFromBinaryFile(data.buffer)
+      tags ? resolve(tags['Orientation']) : reject(new Error('EXIF failed to read exif data'))
     })
   }
 
-  _handleImageLoadSuccess = (src, orientation) => {
+  _handleOrientationSuccess = orientation => {
     if (!this._hasComponentMounted) return
-    this._setOrientation(src, orientation)
+
+    // If there is no Orientation data set for the image, then mark it as null
+    // in the cache to avoid subsequent calls to EXIF
+    if (!orientation || !isNumber(orientation)) {
+      return this._handleOrientationFailure()
+    }
+
+    this._canvasImageTransform(orientation)
   }
 
-  // Don't perform transforms if the image cannot be loaded or there is no EXIF data
-  _handleImgeLoadFailure = src => {
-    _cacheStyleTransforms[src] = NO_TRANSFORM
+  // Mark this image path as no transform and set the ImageComponent src to the original source
+  _handleOrientationFailure = () => {
```

Comment:
maybe log the error here if one comes through?

Reply:
There are a few things that can cause a "failure"
1. `fs.readSync` was not able to read the file
2. `EXIF.getData` returned false because the image is not a jpeg, or there is no EXIF data set on the image. Both of these occur before ever accessing EXIF, so logging here would log for every image type without EXIF
3. A more critical error is getting all the way to reading the EXIF Orientation flags and it is not a value from 1 to 8.